content/guides/java/inject-extract.md

```diff
@@ -76,12 +86,14 @@ public static Scope startServerSpan(Tracer tracer, javax.ws.rs.core.HttpHeaders
 }
 ```
 
-In the above example, instead of using a dedicated adapter class to convert JAXRS `HttpHeaders` type into `io.opentracing.propagation.TextMap`, the headers are copied to a plain `HashMap<String, String>` and converted using a standard adapter `TextMapExtractAdapter`.
+In the above example, instead of using a dedicated adapter class to convert JAX-RS `HttpHeaders` type into `io.opentracing.propagation.TextMap`, the headers are copied to a plain `HashMap<String, String>` and converted using a standard adapter `TextMapExtractAdapter`.
+
+The complete tutorial is available [here](https://github.com/yurishkuro/opentracing-tutorial/tree/master/java/src/main/java/lesson03).
 
 ## Injecting and Extracting TextMap
-The process of injecting and extracting `TextMap` is similar to that of HTTP. Given below are some elaborated code examples of injecting and extracting tracing information using `TextMap` format.
+The process of injecting and extracting `TextMap` is similar to that of HTTP. Given below are some elaborated code examples of [injecting](https://www.programcreek.com/java-api-examples/?code=yangfuhai/jboot/jboot-master/src/main/java/io/jboot/core/rpc/dubbo/JbootDubboConsumerTracingFilter.java) and [extracting](https://www.programcreek.com/java-api-examples/?code=yangfuhai/jboot/jboot-master/src/main/java/io/jboot/core/rpc/dubbo/JbootDubboProviderTracingFilter.java) tracing information using `TextMap` format.
```

Comment:
I have absolutely nothing against this website and author, but don't we have an example of `TextMap` on a repository under the `opentracing-contrib` organization?